app/util/analytics/analytics.py

```diff
@@ -0,0 +1,134 @@
+
+        if not success:
+            message = f"One or more actions have success rate < {SUCCESS_TEST_RATE} %."
+        return success, message
+
+    def is_finished(self):
+        message = 'OK'
+        finished = self.actual_duration >= self.duration
+        if not finished:
+            message = (f"Actual test duration {self.actual_duration} sec "
+                       f"< than expected test_duration {self.duration} sec.")
+        return finished, message
+
+    def is_compliant(self):
+        message = 'OK'
+        compliant = (self.actual_duration >= MIN_DEFAULTS[self.app_type]['test_duration'] and
+                     self.concurrency >= MIN_DEFAULTS[self.app_type]['concurrency'])
+        if not compliant:
+            err_msg = []
+            if self.actual_duration < MIN_DEFAULTS[self.app_type]['test_duration']:
+                err_msg.append(f"Test run duration {self.actual_duration} sec < than minimum test "
+                               f"duration {MIN_DEFAULTS[self.app_type]['test_duration']} sec.")
+            if self.concurrency < MIN_DEFAULTS[self.app_type]['concurrency']:
+                err_msg.append(f"Test run concurrency {self.concurrency} < than minimum test "
+                               f"concurrency {MIN_DEFAULTS[self.app_type]['concurrency']}.")
+            message = ' '.join(err_msg)
+        return compliant, message
+
+    def is_git_operations_compliant(self):
+        # calculate expected git operations for a particular test duration
+        message = 'OK'
+        expected_get_operations_count = int(MIN_DEFAULTS[BITBUCKET]['git_operations_per_hour'] / 3600 * self.duration)
+        git_operations_compliant = self.results_log.actual_git_operations_count >= expected_get_operations_count
+        if not git_operations_compliant:
+            message = (f"Total git operations {self.results_log.actual_git_operations_count} < than "
+                       f"{expected_get_operations_count} - minimum for expected duration {self.duration} sec.")
+        return git_operations_compliant, message
+
+
+def send_analytics(collector: AnalyticsCollector):
+    headers = {"Content-Type": "application/json"}
+    payload = {"run_id": collector.run_id,
+               "user_id": uniq_user_id(),
+               "app_version": collector.application_version,
+               "date": collector.date,
+               "time_stamp": collector.time_stamp,
+               "app_type": collector.app_type,
+               "os": collector.os,
+               "tool_ver": collector.tool_version,
+               "exp_dur": collector.duration,
+               "act_dur": collector.actual_duration,
+               "selenium_test_rates": collector.selenium_test_rates,
+               "jmeter_test_rates": collector.jmeter_test_rates,
+               "concurrency": collector.concurrency
+               }
+    r = requests.post(url=f'{BASE_URL}', json=payload, headers=headers)
+    print(r.json())
```

Comment:
Are You printing 'r' to check the result and You just forgot to delete it or it is needed print?

Reply:
Excellent catch!